src/libraries/System.Memory.Data/src/System/BinaryDataConverter.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace System
+{
+    internal sealed class BinaryDataConverter : JsonConverter<BinaryData>
+    {
+        public sealed override BinaryData? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```

Comment:
consider supporting null tokens and overriding https://docs.microsoft.com/en-us/dotnet/api/system.text.json.serialization.jsonconverter-1.handlenull?view=net-6.0

Reply:
(assuming you want i.e. serialize them to empty binary data)

Reply:
on the second thought that might be unexpected for users 😄